storage_controller/src/service.rs

```diff
@@ -1612,17 +1611,26 @@ impl Service {
         let cancel = CancellationToken::new();
         let reconcilers_cancel = cancel.child_token();
 
+        let mut http_client = reqwest::Client::builder();
+        // We intentionally disable the connection pool, so every request will create its own TCP connection.
+        // It's especially important for heartbeaters to notice more network problems.
```

Comment:
maybe we should make one http client for heartbeaters, and one for everything else? because for non-heartbeat use cases, it makes sense to use pooling IMO.

Reply:
It makes sense, but enabling keep-alive would require some more work done. The problem is that storcon calls `exit(0)` when exiting gracefully, so it doesn't actually destroy all the objects, including HTTP client that keeps a connection pool. As a result, these keep-alive connections are not closed gracefully, and in tests the pageserver's graceful shutdown will hang on trying gracefully close these connections from the server side.

It's possible to fix these issues, but since the problem we are trying to solve now is "parsing certs takes a lot of CPU", I don't think it's worth the trouble for now.

Anyway, I'll write a comment in the code describing these issues, thanks

Reply:
yeah let's improve the comment and not have pooling for now.